pg_shard.c

```diff
@@ -586,6 +607,25 @@ SelectFromMultipleShards(Query *query, List *queryShardList)
 
 
 /*
+ * TODO: Add documentation
+ */
+static Query *
+BuildLocalQuery(Query *query, List *localRestrictList)
+{
+	Query *localQuery = copyObject(query);
+	FromExpr *joinTree = localQuery->jointree;
+
+	if (joinTree != NULL)
```

Comment:
You're right: I was focusing on the `fromlist` having exactly one entry, but we can also expect `joinTree` to never be null. Going to change to assert (don't want to error as this should not be a user-facing problem: `Assert` is more appropriate).